src/form-validation-summary-builder.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ValidationResult,
+  FormValidationSummary,
+  createDefaultFormValidationSummary,
+  FormFieldError,
+} from './model';
+import { arrayContainsEntries } from './helper';
+import { globalFormValidationId } from './const';
+
+const didAllValidationsSucceeded = (
+  validationResults: ValidationResult[]
+): boolean => validationResults.every(fvr => fvr.succeeded);
+
+const extractFieldErrors = (
+  validationResults: ValidationResult[]
+): FormFieldError[] => {
+  const fieldValidationResults = validationResults.filter(
+    fvr => fvr.key !== globalFormValidationId
+  );
+
+  return fieldValidationResults.map(validation => ({
+    key: validation.key,
+    validationResult: validation,
+  }));
+};
+
+const extractFormRecordErrors = (
+  validationResults: ValidationResult[]
+): ValidationResult[] =>
+  validationResults.filter(fvr => fvr.key === globalFormValidationId);
+
+const removeNotValidValidationResults = (
+  validationResults: ValidationResult[]
+): ValidationResult[] => {
+  if (validationResults.some(fvr => fvr === undefined || fvr === null)) {
+    console.error(
+      'Some form validators are returning null / undefined, fix this'
```

Comment:
Maybe include in `console.error` the keys to know which ones?

Reply:
Good point, I will keep the _some_ entry just for performance and if there is an error, I will go for  a filter and display all the faulty keys.

Reply:
Hold on.. we cannot know the key... since the whole validationResult is undefined or null :-)